pilot/pkg/model/config.go

```diff
@@ -95,15 +95,28 @@ func ConfigNamesOfKind(configs sets.Set[ConfigKey], kind kind.Kind) sets.String
 }
 
 // ConfigNamespacedNameOfKind extracts config names of the specified kind.
-func ConfigNamespacedNameOfKind(configs map[ConfigKey]struct{}, kind kind.Kind) map[types.NamespacedName]struct{} {
-	ret := map[types.NamespacedName]struct{}{}
+func ConfigNamespacedNameOfKind(configs map[ConfigKey]struct{}, kind kind.Kind) sets.Set[types.NamespacedName] {
+	ret := sets.New[types.NamespacedName]()
 
 	for conf := range configs {
 		if conf.Kind == kind {
-			ret[types.NamespacedName{
+			ret.Insert(types.NamespacedName{
 				Namespace: conf.Namespace,
 				Name:      conf.Name,
-			}] = struct{}{}
+			})
+		}
+	}
+
+	return ret
+}
+
+// ConfigNameOfKind extracts config names of the specified kind.
+func ConfigNameOfKind(configs map[ConfigKey]struct{}, kind kind.Kind) sets.String {
+	ret := sets.New[string]()
+
+	for conf := range configs {
+		if conf.Kind == kind {
+			ret.Insert(conf.Name)
```

Comment:
donot we need namespace

Reply:
Not for this case, where the configs are not actually Kubernetes types but synthetic `Address` type

Reply:
This is a generic func, if it is only intended for `Address`

We should call `GetAddress` or something else to prevent misuse

Reply:
Its not only intended for Address, its intended anytime you only want `Name`. We just happen to only use that for Addresss